apps/api/src/app/integrations/e2e/get-active-integration.e2e.ts

```diff
@@ -0,0 +1,87 @@
+
+    const { inAppIntegration, emailIntegration, smsIntegration, chatIntegration, pushIntegration } =
+      splitByChannels(activeIntegrations);
+
+    expect(inAppIntegration.length).to.equal(1);
+    expect(emailIntegration.length).to.equal(1);
+    expect(smsIntegration.length).to.equal(1);
+    expect(pushIntegration.length).to.equal(1);
+    expect(chatIntegration.length).to.equal(2);
+
+    expect(inAppIntegration[0].selected).to.equal(true);
+    expect(emailIntegration[0].selected).to.equal(true);
+    expect(smsIntegration[0].selected).to.equal(true);
+    expect(pushIntegration[0].selected).to.equal(true);
+
+    const oneSelected = chatIntegration.some((integration) => integration.selected === true);
+    const oneNotSelected = chatIntegration.some((integration) => integration.selected === false);
+
+    expect(oneSelected).to.equal(true);
+    expect(oneNotSelected).to.equal(true);
+  });
+
+  it('should have thrown an exception if no active integration are exist', async function () {
+    const activeIntegrations = (await session.testAgent.get(`/v1/integrations/active`)).body.data;
+    await deleteAll(activeIntegrations, integrationRepository, session);
+    const response = await session.testAgent.get(`/v1/integrations/active`);
+
+    expect(response.body.statusCode).to.equal(404);
+    expect(response.body.message).to.contain(`No active integrations found for environment ${session.environment._id}`);
+  });
+
+  it('should get active newly created integration with selected flag', async function () {
+    await integrationRepository.create({
+      _environmentId: session.environment._id,
+      _organizationId: session.organization._id,
+      providerId: EmailProviderIdEnum.SES,
+      channel: ChannelTypeEnum.EMAIL,
+      active: true,
+    });
+
+    const activeIntegrations = (await session.testAgent.get(`/v1/integrations/active`)).body.data;
+    const { emailIntegration } = splitByChannels(activeIntegrations);
+
+    expect(emailIntegration.length).to.equal(2);
+
+    const oneSelected = emailIntegration.some((integration) => integration.selected === true);
+    const oneNotSelected = emailIntegration.some((integration) => integration.selected === false);
+
+    expect(oneSelected).to.equal(true);
+    expect(oneNotSelected).to.equal(true);
+  });
+});
+
+function splitByChannels(activeIntegrations) {
+  const inAppIntegration = activeIntegrations.filter((integration) => integration.channel === ChannelTypeEnum.IN_APP);
+  const emailIntegration = activeIntegrations.filter((integration) => integration.channel === ChannelTypeEnum.EMAIL);
+  const smsIntegration = activeIntegrations.filter((integration) => integration.channel === ChannelTypeEnum.SMS);
+  const chatIntegration = activeIntegrations.filter((integration) => integration.channel === ChannelTypeEnum.CHAT);
+  const pushIntegration = activeIntegrations.filter((integration) => integration.channel === ChannelTypeEnum.PUSH);
```

Comment:
Nothing against this, very clear. Though a `reduce` will merge all loops into one.

Reply:
i think it will be a bit less readable but let's make it a bit more optimized 😉